clang-tools-extra/clang-tidy/modernize/UseUniformInitializerCheck.cpp

```diff
@@ -0,0 +1,320 @@
+  const SourceManager &SM = *Result.SourceManager;
+  const ASTContext &Context = *Result.Context;
+  const LangOptions &LangOpts = Context.getLangOpts();
+
+  const auto *VDecl = Result.Nodes.getNodeAs<VarDecl>(VarDeclID);
+  const auto *FDecl = Result.Nodes.getNodeAs<FieldDecl>(FieldDeclId);
+  const auto *CtorInit = Result.Nodes.getNodeAs<CXXCtorInitializer>(CtorInitID);
+
+  // Handle variable declarations
+  if (VDecl != nullptr) {
+    const auto *InitExpr = VDecl->getInit();
+    assert(InitExpr);
+
+    // If the expression is an 'ExprWithCleanups' just get the subexpression
+    if (isa<ExprWithCleanups>(InitExpr))
+      InitExpr = llvm::cast<ExprWithCleanups>(InitExpr)->getSubExpr();
+
+    const VarDecl::InitializationStyle InitStyle = VDecl->getInitStyle();
+
+    // Check for the special case that a default constructor with no arguments
+    // is used
+    // Example: struct A{}; A a;
+    const auto *ConstructorExpr = dyn_cast_or_null<CXXConstructExpr>(InitExpr);
+    if (ConstructorExpr && ConstructorExpr->getNumArgs() == 0)
+      return;
+
+    // Ignore declarations inside a 'CXXForRangeStmt'
+    if (VDecl->isCXXForRangeDecl())
+      return;
+
+    if (InitStyle == VarDecl::InitializationStyle::CInit) {
+      const SourceLocation EqualLoc = utils::lexer::findPreviousTokenKind(
+          InitExpr->getBeginLoc(), SM, LangOpts, tok::equal);
+      if (!EqualLoc.isValid())
+        return;
+
+      auto Diag =
+          diag(EqualLoc,
+               "Use uniform initializer instead of %0-style initializer")
+          << getInitializerStyleName(InitStyle);
+
+      // If we have a C-style initializer with an 'InitListExpr', we only need
+      // to remove the '='
+      // For example: 'int a = {0};'
+      if (isa<InitListExpr>(InitExpr))
+        Diag << FixItHint::CreateRemoval(EqualLoc);
+      else {
+        // Otherwise with a normal c-style initializer we also need to add '{'
+        // and '}'
+        // For example: 'int a = 0;'
+        const std::string ReplacementText =
+            buildReplacementString(InitExpr, Context);
+
+        Diag << FixItHint::CreateReplacement(
+            SourceRange{EqualLoc, InitExpr->getEndLoc()}, ReplacementText);
+      }
+    } else if (InitStyle == VarDecl::InitializationStyle::CallInit) {
+      const SourceRange ParenRange =
+          getParenLocationsForCallInit(InitExpr, SM, LangOpts);
+      if (ParenRange.isInvalid())
+        return;
+
+      const std::string ReplacementString =
+          buildReplacementString(InitExpr, Context);
+
+      diag(ParenRange.getBegin(),
+           "Use uniform initializer instead of %0-style initializer")
+          << getInitializerStyleName(InitStyle)
+          << FixItHint::CreateReplacement(ParenRange, ReplacementString);
+    } else {
+      llvm_unreachable("Invalid initializer style!");
+    }
+  }
+  // Handle field declarations
+  else if (FDecl != nullptr) {
```

Comment:
avoid `if else if else` try change this into bunch of `if (x} { ...; return;}`, or better consider extracting fix generation into separate methods/functions